doc/Programming-guide/Key-platform-features/Endpoint-ownership/index.md

```diff
@@ -168,36 +184,41 @@ The configuration should match the following Avro schema.
-[Creating custom owner verifier]({{root_url}}Customization-guide/Kaa-Server/Development-environment-setup/#creating-custom-user-verifier) page.
+### Custom owner verifier
 
+You can implement plug in your custom verifiers.
+To do this, follow the instructions in [Creating custom owner verifier]({{root_url}}Customization-guide/Kaa-Server/Development-environment-setup/#creating-custom-user-verifier).
 
-# Endpoint access token flow #
-In the *endpoint access token flow*, new endpoints are attached to the owner with the help of the endpoint which was attached to the owner beforehand.
-The following steps illustrate this flow with the endpoint A, which is already attached to the owner, and the endpoint B, which is due to be attached:
 
-   1. The endpoint B periodically generates and sends its access token to the Kaa cluster.
-   2. The endpoint B displays its access token as a QR code on the screen (TV) or on the webpage (e.g., a router or other device with an embedded server).
-   3. The endpoint A retrieves this token by scanning QR code or in any other suitable way and sends it in the endpoint attach request to the Kaa cluster.
-   4. The Kaa cluster verifies the access code and attaches the endpoint B to the owner of the endpoint A.
+## Using the endpoint access token
 
+You can attach new endpoints to an owner using the endpoint that was previously attached to that owner.
+Assume we have a new endpoint B that we want to attach to the owner using the previously attached endpoint A.
 
-![Owner access token flow](endpoint_attach_2.png "Owner access token flow")
+The process flows as follows:
+
+1. The endpoint B periodically generates and sends its access token to the Kaa cluster.
+
+2. The endpoint B displays its access token as a QR code on the screen (TV) or on the webpage (e.g., a router or other device with an embedded server).
 
+3. The endpoint A retrieves this token by scanning QR code or in any other suitable way and sends it in the endpoint attach request to the Kaa cluster.
 
-## Attach endpoint to owner ##
+4. The Kaa cluster verifies the access code and attaches the endpoint B to the owner of the endpoint A.
 
-To enable sending/receiving events to/from endpoints, at first the client should attach the endpoint to the owner as shown in the following examples for different platforms SDK.
 
-> Please refer to [Glossary]({{root_url}}Glossary/) for more details about **userExternalId**, **userAccessToken**, **userVerifierToken** 
->and other parameters which are used by KaaClient methods.
+![Owner access token flow](endpoint_attach_2.png "Owner access token flow")
+
+
+### Attach endpoint to owner
+
+To enable sending/receiving events to/from endpoints, [Kaa client]({{root_url}}Glossary/#kaa-client) first attaches the endpoint to the owner as shown in the following examples for different [SDK types]({{root_url}}Glossary/#sdk-type).
 
 <ul class="nav nav-tabs">
```

Comment:
Sample code seems to be misplaced. There must be sample code snippets in both active and assisted modes for attaching and detaching endpoints from owners.